lib/containers/common.pm

```diff
@@ -105,18 +105,27 @@ sub install_docker_when_needed {
     }
     systemctl('is-active docker');
     systemctl('status docker', timeout => 120);
-    assert_script_run('docker info');
+    record_info('docker', script_output('docker info'));
 }
 
 sub install_buildah_when_needed {
     my $host_os = shift;
-    my @pkgs = qw(buildah);
     if (script_run("which buildah") != 0) {
-        # We may run openSUSE with DISTRI=sle and opensuse doesn't have SUSEConnect
-        activate_containers_module if $host_os =~ 'sles';
-        zypper_call "in @pkgs";
-        record_info('buildah', script_output('buildah info'));
+        if ($host_os eq 'centos') {
+            assert_script_run "dnf -y install buildah", timeout => 300;
+        } elsif ($host_os eq 'ubuntu') {
+            my $version_id = script_output('(. /etc/os-release && echo $VERSION_ID)');
+            my $ubuntu_repo = "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${version_id}";
```

Comment:
Neat picking, bash variable `$VERSION_ID` could be possibly used in the link if the subshell invocation on L#117 is removed.
In such a case, the L#117 could be just an `assert_script_call '.  /etc/os-release'`

Reply:
I will actually modify the Ubuntu image to come with this repo in place already, no need to do it here, and also, we don't need to repeat this code in `install_podman_when needed` and here. Moving to `DRAFT` until that is done.